internal/service/vpc/resource_aws_vpc_peering_connection.go

```diff
@@ -0,0 +1,337 @@
+	if projectName == "" || vpcID == "" {
+		return diag.Errorf("incorrect VPC ID, expected structure <PROJECT_NAME>/<VPC_ID>")
+	}
+
+	awsAccountId := d.Get("aws_account_id").(string)
+	awsVPCId := d.Get("aws_vpc_id").(string)
+	awsVPCRegion := d.Get("aws_vpc_region").(string)
+	if awsVPCRegion != "" {
+		region = &awsVPCRegion
+	}
+
+	if _, err = client.VPCPeeringConnections.Create(
+		projectName,
+		vpcID,
+		aiven.CreateVPCPeeringConnectionRequest{
+			PeerCloudAccount: awsAccountId,
+			PeerVPC:          awsVPCId,
+			PeerRegion:       region,
+		},
+	); err != nil {
+		return diag.Errorf("Error waiting for AWS VPC peering connection creation: %s", err)
+	}
+
+	stateChangeConf := &resource.StateChangeConf{
+		Pending: []string{"APPROVED"},
+		Target: []string{
+			"ACTIVE",
+			"REJECTED_BY_PEER",
+			"PENDING_PEER",
+			"INVALID_SPECIFICATION",
+			"DELETING",
+			"DELETED",
+			"DELETED_BY_PEER",
+		},
+		Refresh: func() (interface{}, string, error) {
+			pc, err := client.VPCPeeringConnections.GetVPCPeering(
+				projectName,
+				vpcID,
+				awsAccountId,
+				awsVPCId,
+				region,
+			)
+			if err != nil {
+				return nil, "", err
+			}
+			return pc, pc.State, nil
+		},
+		Delay:      10 * time.Second,
+		Timeout:    d.Timeout(schema.TimeoutCreate),
+		MinTimeout: 2 * time.Second,
+	}
+
+	res, err := stateChangeConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.Errorf("Error creating VPC peering connection: %s", err)
+	}
+
+	diags := diag.Diagnostics{}
+	pc = res.(*aiven.VPCPeeringConnection)
+	if pc.State != "ACTIVE" {
+		switch pc.State {
+		case "PENDING_PEER":
+			diags = append(diags, diag.Diagnostic{
+				Severity: diag.Warning,
+				Summary: fmt.Sprintf("Aiven platform has created a connection to the specified "+
+					"peer successfully in the cloud, but the connection is not active until the user "+
+					"completes the setup in their cloud account. The steps needed in the user cloud "+
+					"account depend on the used cloud provider. Find more in the state info: %s",
+					stateInfoToString(pc.StateInfo)),
+			})
+		case "DELETED":
+			diags = append(diags, diag.Errorf("A user has deleted the peering connection through the Aiven "+
+				"Terraform provider, or Aiven Web Console or directly via Aiven API. There are no "+
+				"transitions from this state")...)
+		case "DELETED_BY_PEER":
+			diags = append(diags, diag.Errorf("A user deleted the peering cloud resource in their account. "+
+				"There are no transitions from this state")...)
+		case "REJECTED_BY_PEER":
+			diags = append(diags, diag.Errorf("AWS VPC peering connection request was rejected, state info: %s",
+				stateInfoToString(pc.StateInfo))...)
+		case "INVALID_SPECIFICATION":
+			diags = append(diags, diag.Errorf("VPC peering connection cannot be created, more in the state info: %s",
+				stateInfoToString(pc.StateInfo))...)
+		default:
+			return diag.Errorf("Unknown VPC peering connection cannot state: %s", pc.State)
```

Comment:
Probably should be `Unknown VPC peering connection state` instead?

Reply:
:+1: